src/components/Dropdown/Items/MultiSelectItem/MultiSelectItem.tsx

```diff
@@ -3,57 +3,78 @@ import {
     DropdownMenuCheckboxItem,
     DropdownMenuCheckboxItemProps,
 } from '@radix-ui/react-dropdown-menu';
+import { PrioritySelector, PrioritySelectorProps } from '../../../PrioritySelector';
 import { bem } from '../../../../utils';
 import { VisualCheckbox } from '../../../Checkbox';
 import { Text } from '../../../Text';
 import styles from './MultiSelectItem.scss';
+import { stopPropagation } from '../../../../utils/misc';
 
-export interface Props extends DropdownMenuCheckboxItemProps {
+export interface Props<PriorityItemValue> extends DropdownMenuCheckboxItemProps {
     /** Id for the checkbox */
     id?: string;
     /** a variant to determine the look and feel of component */
     variant?: 'option' | 'select-all' | 'group-title';
-    /** to add a priority badge before the label */
-    hasPriority?: boolean;
+    /** props for PrioritySelector  */
+    priority?: PrioritySelectorProps<PriorityItemValue>;
     /** Checkbox status */
     isSelected?: boolean;
 }
 
 const { block } = bem('MultiSelectItem', styles);
 
-export const MultiSelectItem = React.forwardRef<HTMLLIElement, Props>(
+export const MultiSelectItem = React.forwardRef<HTMLLIElement, Props<string>>(
```

Comment:
this restricts the ItemValue option. At first I thought I'll just leave it, but now I had the same problem for Enhanced Button and found [this](https://stackoverflow.com/questions/58469229/react-with-typescript-generics-while-using-react-forwardref). I used the first solution there to implement it correctly. If the code there is confusing (it was for me at first) we can have a call and I show you how it worked for me.

Same for the SingleSelectItem